### ISSUE TITLE
Change MSD size to match real size

### ISSUE THREAD

**Issue by tusker-tools** [sfyip/STM32F103_MSD_BOOTLOADER#5]
Currently, the total size of the mass storage device shows 116MB, from which 115MB are free (according Windows Explorer).

It would be better to display the actual sizes:
- Total Memory = DEV_FLASH_SIZE
- Free Memory = APP_SIZE

Furthermore, it would be better to define a new compiler constant for the bootloader size and use it for calculating APP_ADDR and APP_SIZE in btldr_config.h

**Comment by sfyip** [comment]
The minimum FAT32 disk size is 32MB, it cannot indicated 64KB flash capacity.

**Comment by tusker-tools** [comment]
You are right about the minimum capacity.

Then could we at least have a total volume of 32MB where only 64KB minus BOOTLOADER_SIZE is free and the rest is occupied?

**Comment by tusker-tools** [comment]
Do you have the information how e.g. Windows calculates the free space of a FAT32 partition?
Unfortunately I can't find anything in the internet, so I don't know how to realize it.

**Comment by sfyip** [comment]
http://elm-chan.org/fsw/ff/doc/getfree.html

**Comment by oliinykb** [comment]
Thus, have you succeeded with the volume size correction?

**Comment by tusker-tools** [comment]
No, I did not succeed. I tried different configuration parameters for the FAT32 partition, but this did not lead to the intended change. Unfortunately, I'm not deep enough into the FAT32 internals. I would appreciate if anybody could help.
